### PR TITLE
#2927 Add staticRefs for Bus, Line, TransformerFixedRatio and Infinit…

### PR DESCRIPTION
**Pull request by gautierbureau**
…eBus.

**Checklist before requesting a review**

*use `'[x]'` to check the checkboxes, or submit the PR and then click the checkboxes*

- [ ] unit tests and non-regression tests were added (new model, new feature and bug fix)
- [ ] main documentation was updated (update of input/output file, 3rd party, model, repository organization, solver)
- [ ] example documentations were updated (new example in examples folder)
- [ ] the corresponding milestone was added in the ticket and in this PR
- [ ] if this PR modifies the parameters or inputs/outputs of a model/solver: the corresponding xsl was added in util/xsl and platform DB were updated
- [ ] if this PR modifies a dictionary: the corresponding french dictionary was updated

### CODE REVIEW

File: dynawo/sources/Models/Modelica/Dynawo/Electrical/Buses/Bus.mo [hunk 1]
```diff
@@ -20,10 +20,14 @@ model Bus "Bus"
 
   Connectors.ACPower terminal annotation(
     Placement(visible = true, transformation(origin = {-1.42109e-14, 98}, extent = {{-10, -10}, {10, 10}}, rotation = 0), iconTransformation(origin = {-1.42109e-14, 0}, extent = {{-10, -10}, {10, 10}}, rotation = 0)));
+  Types.VoltageModulePu UPu "Voltage amplitude at terminal in pu (base UNom)";
+  Types.Angle phase "Phase at terminal";
```
Comment (cossartque) [inline dynawo/sources/Models/Modelica/Dynawo/Electrical/Buses/Bus.mo:24]: UPhase "Voltage angle at terminal in rad"

File: dynawo/sources/Models/Modelica/Dynawo/Electrical/Buses/InfiniteBus.mo [hunk 2]
```diff
@@ -28,9 +28,14 @@ model InfiniteBus "Infinite bus"
   parameter Types.PerUnit UPu "Infinite bus constant voltage module";
   parameter Types.Angle UPhase "Infinite bus constant voltage angle";
 
+  Types.VoltageModulePu UPuVar "Voltage amplitude at terminal in pu (base UNom)";
+  Types.Angle phase "Phase at terminal";
```
Comment (cossartque) [inline dynawo/sources/Models/Modelica/Dynawo/Electrical/Buses/InfiniteBus.mo:32]: UPhase "Voltage angle at terminal in rad"

File: dynawo/sources/Models/Modelica/Dynawo/Electrical/Transformers/TransformerFixedRatio.mo [hunk 3]
```diff
@@ -39,6 +39,11 @@ model TransformerFixedRatio "Two winding transformer with a fixed ratio"
 
   parameter Types.PerUnit rTfoPu "Transformation ratio in pu: U2/U1 in no load conditions";
 
+  Types.ActivePowerPu P1Pu "Active power on side 1 in pu (base SnRef)";
```
Comment (cossartque) [inline dynawo/sources/Models/Modelica/Dynawo/Electrical/Transformers/TransformerFixedRatio.mo:42]: in pu (base SnRef) (receptor convention)

For all P and Q here

File: dynawo/sources/Models/Modelica/Dynawo/Electrical/Transformers/TransformerFixedRatio.mo [hunk 4]
```diff
@@ -49,6 +54,11 @@ equation
     terminal2.i = Complex (0);
   end if;
 
+  P1Pu = ComplexMath.real(terminal1.V * ComplexMath.conj(terminal1.i));
```
Comment (cossartque) [inline dynawo/sources/Models/Modelica/Dynawo/Electrical/Transformers/TransformerFixedRatio.mo:57]: Already exists for the lines ?

**Review by cossartque — CHANGES_REQUESTED** [review]

File: dynawo/sources/Models/Modelica/Dynawo/Electrical/Transformers/TransformerFixedRatio.mo (hunk 4 above)
Comment (gautierbureau) [reply dynawo/sources/Models/Modelica/Dynawo/Electrical/Transformers/TransformerFixedRatio.mo:57]: yes

**Review by cossartque — DISMISSED** [review]

File: dynawo/sources/Modeler/DataInterface/PowSyblIIDM/test/TestDataInterface.cpp [hunk 5]
```diff
@@ -593,6 +595,8 @@ createBusBreakerNetwork(const BusBreakerNetworkProperty& properties) {
           .setRho(1.)
           .endStep()
           .add();
+      transformer.getLeg1().getRatioTapChanger().setRegulationTerminal(stdcxx::ref(transformer.getLeg1().getTerminal()));
```
Comment (rosiereflo) [inline dynawo/sources/Modeler/DataInterface/PowSyblIIDM/test/TestDataInterface.cpp:598]: Does not change the tests?

File: dynawo/sources/Models/CPP/ModelNetwork/DYNModelNetwork.cpp [hunk 6]
```diff
@@ -443,7 +443,10 @@ ModelNetwork::initializeFromData(const shared_ptr<DataInterface>& data) {
     data->setReference("p2", id, id, "P2_value");
     data->setReference("q2", id, id, "Q2_value");
     data->setReference("state", id, id, "state_value");
-    data->setReference("tapIndex", id, id, "step_value");
+    if (modelTwoWindingsTransformer->getModelRatioTapChanger() || modelTwoWindingsTransformer->getModelPhaseTapChanger())
```
Comment (rosiereflo) [inline dynawo/sources/Models/CPP/ModelNetwork/DYNModelNetwork.cpp:446]: Why 2 if and not a or?

File: dynawo/sources/Models/CPP/ModelNetwork/DYNModelTapChanger.h [hunk 7]
```diff
@@ -165,12 +166,26 @@ class ModelTapChanger {
    */
   inline void setTNext(double time) { tNext_ = time; }
 
+  /**
+   * @brief set fictitious property
+   * time
+   * @param time time
```
Comment (rosiereflo) [inline dynawo/sources/Models/CPP/ModelNetwork/DYNModelTapChanger.h:172]: wrong comment

File: util/envDynawo.sh [hunk 8]
```diff
@@ -987,7 +987,7 @@ build_tests() {
 
   tests=$@
   if [ -z "$tests" ]; then
-    cmake --build $DYNAWO_BUILD_DIR --target tests --config Debug
+    cmake --build $DYNAWO_BUILD_DIR --target tests --config Debug -j 1
```
Comment (rosiereflo) [inline util/envDynawo.sh:990]: why?

**Review by rosiereflo — CHANGES_REQUESTED** [review]

File: util/envDynawo.sh (hunk 8 above)
Comment (gautierbureau) [reply util/envDynawo.sh]: wrong add to the commit

File: dynawo/sources/Models/CPP/ModelNetwork/DYNModelNetwork.cpp (hunk 6 above)
Comment (gautierbureau) [reply dynawo/sources/Models/CPP/ModelNetwork/DYNModelNetwork.cpp:446]: no reason I just added the else if after a first version so I did not harmonized, I can do only one if

File: dynawo/sources/Modeler/DataInterface/PowSyblIIDM/test/TestDataInterface.cpp (hunk 5 above)
Comment (gautierbureau) [reply dynawo/sources/Modeler/DataInterface/PowSyblIIDM/test/TestDataInterface.cpp:598]: I will have another look at it, maybe it is not needed anymore

**Review by cossartque — DISMISSED** [review]

**Review by cossartque — APPROVED** [review]

**Review by rosiereflo — APPROVED** [review]

**Pull request merged**